### PR TITLE
validations and model specs 

### PR DESCRIPTION
**Pull request by Progress2002**
### The following were implemented in this milestone:
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
 RSpec.describe Comment, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  subject { Comment.new(text: 'Nice') }
+
+  before { subject.save }
+
+  it 'Comments text should contain text' do
+    subject.text = nil
+    expect(subject).to_not be_valid
+  end
```
Comment (Kamba56) [inline spec/models/comment_spec.rb:48]: - [x]  Great job adding unit test for comments model. As demanded by project requirements, kindly add unit test for all custom model methods. In the case of the comments model, add unit test for `update_comment_counter`.

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
 
 RSpec.describe Like, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  subject { Like.new }
+
+  before { subject.save }
+
+  it 'likes text should contain text' do
+    expect(subject).to_not be_valid
+  end
```
Comment (Kamba56) [inline spec/models/like_spec.rb:44]: - [x]  Great job adding unit test for the likes model. As demanded by project requirements, kindly add unit tests for all custom model methods. In the case of the likes model, add unit test for `update_like_counter`.

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -1,5 +1,72 @@
...
+require_relative '../rails_helper'
 
 RSpec.describe Post, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  let(:user) { User.create(name: 'Fuad', posts_counter: 0) }
+  subject do
+    Post.new(user_id: user.id, title: 'Hello', text: 'This is my post', comment_counter: 1, likes_counter: 1)
+  end
+
+  # before { subject.save }
```
Comment (Kamba56) [inline spec/models/post_spec.rb]: - [x] Kindly remove all commented pieces of code from your documents. This is to ensure a clean and professional document.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @Progress2002, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job adding validations for posts and users model!👏👏👏
- [x] Great job using rspec to add unit test for models

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 3 above)
Comment (Progress2002) [reply spec/models/post_spec.rb]: Hello @Kamba56, thanks for the review. The project has been updated to meet the requirement as stated above.

File: spec/models/like_spec.rb (hunk 2 above)
Comment (Progress2002) [reply spec/models/like_spec.rb:44]: Hello @Kamba56, thanks for the review. The project has been updated to meet the requirement as stated above.

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (Progress2002) [reply spec/models/comment_spec.rb:48]: Hello @Kamba56, thanks for the review. The project has been updated to meet the requirement as stated above.

**Review by OyePriscilla — APPROVED** [review]
Hi @Progress2002 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Implemented test for comment and like counters ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**